docs/source/kernel-spark-standalone.md

```diff
@@ -0,0 +1,142 @@
+
+Having said that, even if you are not leveraging the full distributed capabilities of Jupyter Enterprise Gateway, client mode can still help mitigate resource starvation by enabling a pseudo-distributed mode, where kernels are started in different nodes of the cluster utilizing a round-robin algorithm. In this case, you can still experience bottlenecks on a given node that receives requests to start "large" kernels, but otherwise, you will be better off compared to when all kernels are started on a single node or as local processes, which is the default for vanilla Jupyter Notebook.
+
+The pseudo-distributed capabilities are currently supported in Spark Standalone and require the following environment variables to be set:
+
+* SPARK_HOME: Must point to the Apache Spark installation path
+
+```
+SPARK_HOME:/usr/hdp/current/spark2-client                            #For HDP distribution
+```
+
+* EG_REMOTE_HOSTS must be set to a comma-separated set of FQDN hosts indicating the hosts available for running kernels. (This can be specified via the command line as well: `--EnterpriseGatewayApp.remote_hosts`)
+
+```
+EG_REMOTE_HOSTS=elyra-node-1.fyre.ibm.com,elyra-node-2.fyre.ibm.com,elyra-node-3.fyre.ibm.com,elyra-node-4.fyre.ibm.com,elyra-node-5.fyre.ibm.com
+```
+
+### Configuring Kernels for Spark Standalone
+
+Although Enterprise Gateway does not currently provide sample kernelspecs for Spark standalone, here are the steps necessary to convert a yarn_client kernelspec to standalone.
+
+For each supported Jupyter Kernel, we have provided sample kernel configurations and launchers as part of the release
+[jupyter_enterprise_gateway_kernelspecs-2.0.0.dev2.tar.gz](https://github.com/jupyter/enterprise_gateway/releases/download/v2.0.0rc1/jupyter_enterprise_gateway_kernelspecs-2.0.0rc1.tar.gz).
+
+Considering we would like to enable the IPython Kernel that comes pre-installed with Anaconda to run on
+Spark Standalone, we would have to copy the sample configuration folder **spark_python_yarn_client**
+to where the Jupyter kernels are installed (e.g. jupyter kernelspec list) and rename it to **spark_python_spark_standalone***
+
+``` Bash
+wget https://github.com/jupyter/enterprise_gateway/releases/download/v2.0.0rc1/jupyter_enterprise_gateway_kernelspecs-2.0.0rc1.tar.gz
+SCALA_KERNEL_DIR="$(jupyter kernelspec list | grep -w "python3" | awk '{print $2}')"
+KERNELS_FOLDER="$(dirname "${SCALA_KERNEL_DIR}")"
+tar -zxvf enterprise_gateway_kernelspecs.tar.gz --strip 1 --directory $KERNELS_FOLDER/spark_python_yarn_client/ spark_python_yarn_client/
+mv $KERNELS_FOLDER/spark_python_yarn_client $KERNELS_FOLDER/spark_python_spark_standalone
+```
+
+After that, you should have a kernel.json that looks similar to the one below:
```

Comment:
I think we need to mention the edits that need to occur to convert a yarn-client kernel.json file to standalone prior to this statement.

The edits are to update the `display_name` (however they like, your text is great) and to update the `--master` option in the `SPARK_OPTS` to point to the spark master node rather than indicate `--deploy-mode client`.

I think we would probably drop off the `spark.yarn.submit.waitAppCompletion` option as well, although I'm not certain about that.  (@lresende - is `--conf spark.yarn.submit.waitAppCompletion` still necessary in standalone mode?)

After THAT, they should see a kernel.json ... :smiley:

Reply:
I have removed spark.yarn.submit.waitAppCompletion (it is about yarn, not spark cluster), I can add it back if @lresende says otherwise.